client/my-sites/controller.js

```diff
@@ -552,9 +552,21 @@ export function siteSelection( context, next ) {
 		// onSelectedSiteAvailable might render an error page about domain-only sites or redirect
 		// to wp-admin. In that case, don't continue handling the route.
 		dispatch( setSelectedSiteId( siteId ) );
-		if ( onSelectedSiteAvailable( context ) ) {
-			next();
-		}
+
+		const currentPlanSlug = getSitePlanSlug( getState(), siteId );
+		const shouldUpdateStateAfterUpgrade =
+			context.pathname.startsWith( '/plans/my-plan/trial-upgraded/' ) &&
+			[ PLAN_FREE, PLAN_JETPACK_FREE ].includes( currentPlanSlug );
+
+		// This will fetch the site and update the state after the plan is upgraded if the site is on the trial-upgraded flow.
+		const promise = shouldUpdateStateAfterUpgrade
+			? dispatch( requestSite( siteId ) )
```

Comment:
The Trial Upgraded page will be rendered as it should. However, it may be worth invalidating the cache for the upgraded site.

<img width="1508" alt="Screen Shot 2023-03-29 at 11 47 09" src="https://user-images.githubusercontent.com/1234758/228577121-ea269d11-daf0-4d71-8cd1-1bcf015c71e0.png">

Reply:
Sounds good. We might also catch the error just in case.